### ISSUE TITLE
Support String Catalog (.xcstrings)

### ISSUE THREAD

**Issue by Zomatree** [WeblateOrg/weblate#10848]
### Describe the problem

Weblate only currently supports the legacy strings file and strings dict, and not the new string catalog file

### Describe the solution you would like

Support the new format

### Describe alternatives you have considered

I have considered switching to the legacy format but that is not a long term solution

### Screenshots

![CleanShot 2024-01-24 at 15 26 24](https://github.com/WeblateOrg/weblate/assets/39768508/ecd5ba18-f1d8-4a2e-b498-6eb5e9cc641f)

**Comment by nijel** [comment]
Is there any documentation available on the format?

**Comment by nijel** [comment]
While looking at format examples (for https://github.com/translate/translate/issues/5190), I've realized that this is a multi-language file and that would also need some changes in the Weblate core: https://github.com/WeblateOrg/weblate/issues/11039

**Comment by libewa** [comment]
Quick tip: In the meantime you can also also Export Localizations from Xcode's Product menu. But iirc you can't automate this, and using Weblate should be as automated as possible in my opinion.

**Comment by libewa** [comment]
This feature exports the `.xcstrings` files to `.xcloc` archives, which are just folders containing XLIFF files.

**Comment by kewisch** [comment]
@nijel We'd like to use xcstrings for Thunderbird for iOS. Is there anything additionally we can do to support development on this? It sounds like with the necessary changes to weblate this would be quite a heavy lift to properly support?

**Comment by kewisch** [comment]
Absolutely, very frustrating they aren't sharing this from the start. Using external tooling is such a common use case, though i guess they don't want to deal with the backwards compatibility just yet. I hope we can still make it work. Is 11039 something your team is considering to work on soon?

(I'm also happy to move this conversation into email or chat if you prefer, I want to be mindful that some parts may end up being off topic for this specific issue)

**Comment by nijel** [comment]
@kewisch It's currently not on our roadmap, https://github.com/WeblateOrg/weblate/issues/11039#issuecomment-1952347300 is still valid.

**Comment by marcprux** [comment]
I think `.xcstrings` support will become more and more important over time, as this is the default path that Xcode is guiding iOS/macOS developers down for translations. The older "Strings File" format is now marked as "Legacy".

<img width="1325" height="862" alt="Image" src="https://github.com/user-attachments/assets/ae44fc5a-1cfc-4e00-84d5-789eca47ec0b" />

**Comment by nijel** [comment]
If you have any input on the file format, such as vendor specification, please provide it at https://github.com/translate/translate/issues/5190.
